Jenkinsfile

```diff
@@ -0,0 +1,56 @@
+pipeline {
+    agent{
+        label 'jenkins-workers'
+    }
+
+    options {
+        timestamps()
+        buildDiscarder(logRotator(numToKeepStr: "10"))
+    }
+
+    environment {
+        BUILD_TAG = sh label: 'Generating build tag', returnStdout: true, script: 'python3 scripts/tag.py ${GIT_BRANCH} ${BUILD_NUMBER} ${GIT_COMMIT}'
+        BUILD_TAG_LOWER = sh label: 'Lowercase build tag', returnStdout: true, script: "echo -n ${BUILD_TAG} | tr '[:upper:]' '[:lower:]'"
+        ENVIRONMENT_ID = "build1"
+        ECR_REPO_DIR = "gp2gp"
+        DOCKER_IMAGE = "${DOCKER_REGISTRY}/${ECR_REPO_DIR}:${BUILD_TAG}"
+    }
+
+    stages {
+        stage('Build') {
+            stages {
+                stage('Build Docker Images') {
```

Comment:
Can we add a test stage? Looks like there is one test in `main` so it should work. Would need to add a Dockerfile.test as was done in 111

Reply:
Any reason why tests should not be integral part of container build process?

Reply:
Never mind, I think I have found compromise there.

Reply:
Apparently due to how jenkins and docker cache works we're wasting time by running tests twice - http://ec2-3-9-114-173.eu-west-2.compute.amazonaws.com/job/integration-adaptor-gp2gp/view/change-requests/job/PR-10/.

Do you really need separate test stage?